openwisp_firmware_upgrader/admin.py

```diff
@@ -102,9 +102,9 @@ def upgrade_selected(self, request, queryset):
             {
                 'title': title,
                 'related_device_fw': related_device_fw,
-                'related_count': len(related_device_fw),
+                'related_count': related_device_fw.count(),
                 'firmwareless_devices': firmwareless_devices,
-                'firmwareless_count': len(firmwareless_devices),
+                'firmwareless_count': firmwareless_devices.count(),
```

Comment:
both querysets would be evaluated anyway during template rendering to build the list of devices, hence the full query will be done anyway.
So with this change, we're adding 2 more queries.
Or am I missing something?

Reply:
As anticipated, I added an assertion for the number of queries in the tests to test whether my assumption was correct.

Reply:
Well spotted, I've removed that part